fluent/src/commonMain/kotlin/com/konyaco/fluent/component/CalendarView.kt

```diff
@@ -0,0 +1,681 @@
+        color = FluentTheme.colors.background.layer.default,
+        border = null
+    ) {
+        Column(Modifier.padding(4.dp)) {
+            // Day of Weeks
+            Row(
+                modifier = Modifier.fillMaxWidth(),
+                horizontalArrangement = Arrangement.SpaceAround
+            ) {
+                val weekNames = state.dayOfWeekNames.value
+                val start = state.localeStartDayOfWeek - 1
+
+                for (i in weekNames.indices) {
+                    val j = (start + i) % weekNames.size
+                    DaysOfTheWeek(weekNames[j])
+                }
+            }
+
+            LazyVerticalGrid(
+                GridCells.Fixed(7),
+                modifier = Modifier.height(250.dp).fillMaxWidth(),
+                verticalArrangement = Arrangement.spacedBy(2.dp),
+                contentPadding = PaddingValues(2.dp)
+            ) {
+                itemsIndexed(state.candidateDays.value) { i, e ->
+                    Box(contentAlignment = Alignment.Center) {
+                        val current = e == state.currentDay.value
+                        val selected = e == state.selectedDay.value
+                        val outOfRange = e.monthValue != state.viewMonth.value.monthValue
+                        Item(
+                            text = e.day.toString(),
+                            size = ItemSize.Day,
+                            header = if (e.day == 1) state.monthNames.value[e.monthValue % 12] else null,
+                            current = current,
+                            selected = selected,
+                            blackOut = false, // TODO: Support blackOut later
+                            outOfRange = outOfRange,
+                            onSelectedChange = {
+                                state.selectDay(e)
+                                onChoose(e)
+                            }
+                        )
+                    }
+                }
+            }
+        }
+    }
+}
+
+private enum class ItemSize {
+    Day, MonthYear
+}
+
+@Composable
+private fun CalendarHeader(
+    modifier: Modifier,
+    text: String,
+    disabled: Boolean,
+    onClick: () -> Unit
+) {
+    Box(modifier) {
+        SubtleButton(onClick = onClick, iconOnly = true, disabled = disabled) {
+            Box(
+                modifier = Modifier.width(202.dp).padding(horizontal = 12.dp),
+                contentAlignment = Alignment.CenterStart
+            ) {
+                Text(text = text, style = FluentTheme.typography.bodyStrong)
+            }
+        }
+    }
+}
+
+@Composable
+private fun DaysOfTheWeek(text: String) {
+    Box(Modifier.size(38.dp), contentAlignment = Alignment.Center) {
+        Text(text = text, style = FluentTheme.typography.bodyStrong)
+    }
+}
+
+
+@Composable
+private fun PaginationButton(
+    up: Boolean,
+    onClick: () -> Unit
+) {
+    SubtleButton(onClick = onClick, iconOnly = true) {
+        if (up) Icon(
```

Comment:
move the statement switch to `imageVector = `, which can reuse the `icon` and `Modifier` component:
```kotlin
Icon(
    imageVector = if (up) Icons.Filled.CareUp else Icons.Filled.CaretDown,
    contentDescrpition = if (up) "Up" else "Down",
    modifier = Modifier.size(12.dp),
)
```